libraries/AP_Airspeed/AP_Airspeed_AUAV.cpp

```diff
@@ -0,0 +1,246 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+  backend driver for airspeed from a I2C AUAV sensor
+ */
+#include "AP_Airspeed_AUAV.h"
+
+#if AP_AIRSPEED_AUAV_ENABLED
+
+#define AUAV_AIRSPEED_I2C_ADDR 0x26
+
+// the sensor supports a 2nd channel (2nd I2C address) for absolute pressure
+// we could use this as a baro driver but don't yet
+#define AUAV_AIRSPEED_I2C_ADDR_ABSOLUTE 0x27
+
+// max measurement time for 8x averaging differential is 16.2ms
+#define MEASUREMENT_TIME_MAX_MS 17
+
+#define MEASUREMENT_TIMEOUT_MS 200
+
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+#include <GCS_MAVLink/GCS.h>
+
+extern const AP_HAL::HAL &hal;
+
+
+AP_Airspeed_AUAV::AP_Airspeed_AUAV(AP_Airspeed &_frontend, uint8_t _instance, const float _range_inH2O) :
+    AP_Airspeed_Backend(_frontend, _instance),
+    range_inH2O(_range_inH2O)
+{
+}
+
+// probe for a sensor
+bool AP_Airspeed_AUAV::probe(uint8_t bus, uint8_t address)
+{
+    _dev = hal.i2c_mgr->get_device(bus, address);
+    if (!_dev) {
+        return false;
+    }
+
+    WITH_SEMAPHORE(_dev->get_semaphore());
+
+    _dev->set_retries(10);
+    _measure();
+    hal.scheduler->delay(10);
+    _collect();
+
+    return last_sample_time_ms != 0;
+
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_AUAV::init()
+{
+    if (bus_is_configured()) {
+        // the user has configured a specific bus
+        if (probe(get_bus(), AUAV_AIRSPEED_I2C_ADDR)) {
+            goto found_sensor;
+        }
+    } else {
+        // if bus is not configured then fall back to the old
+        // behaviour of probing all buses, external first
+        FOREACH_I2C_EXTERNAL(bus) {
+            if (probe(bus, AUAV_AIRSPEED_I2C_ADDR)) {
+                goto found_sensor;
+            }
+        }
+        FOREACH_I2C_INTERNAL(bus) {
```

Comment:
I wonder if we should actually have a force-probe-internal-buses we can enable on those boards which screw this up?  Might save flash on boards which are not stuffed...

Reply:
i've changed it to require the bus to be set